charmcraft/commands/build.py

```diff
@@ -344,7 +345,7 @@ def pack_charm_in_instance(
             "(may take a while the first time but it's reusable)"
         )
 
-        build_base = providers.BASE_CHANNEL_TO_PROVIDER_BASE[build_on.channel]
+        build_base_alias = get_base_alias((build_on.name, build_on.channel))
```

Comment:
Thinking back, I think we could have made this function take the name and channel as separate parameters, rather than a tuple.  Oh well.

Reply:
And convert to tuple in the function? Maybe more flexible.